src/status_im2/contexts/chat/messages/list/view.cljs

```diff
@@ -102,47 +103,69 @@
+       [quo/floating-shell-button
+        (merge {:jump-to
+                {:on-press #(do
+                              (rf/dispatch [:chat/close true])
+                              (rf/dispatch [:shell/navigate-to-jump-to]))
+                 :label    (i18n/label :t/jump-to)}}
+               (when @show-floating-scroll-down-button
+                 {:scroll-to-bottom {:on-press scroll-to-bottom}}))
+        {:position :absolute
+         :bottom   (+ (:bottom insets) composer.constants/composer-default-height 6)}]])))
+
 (defn messages-list
-  [{:keys [chat-id] :as chat}]
-  (let [context        (rf/sub [:chats/current-chat-message-list-view-context])
-        messages       (rf/sub [:chats/raw-chat-messages-stream chat-id])
-        keyboard-shown (atom false)
-        bottom-space   15]
-    [rn/view
-     {:style {:flex 1}}
-     ;; NOTE: DO NOT use anonymous functions for handlers
-     [rn/flat-list
-      {:key-fn                       list-key-fn
-       :ref                          list-ref
-       :header                       [list-header chat]
-       :footer                       [list-footer chat]
-       :data                         messages
-       :render-data                  {:context        context
-                                      :keyboard-shown keyboard-shown}
-       :render-fn                    render-fn
-       :on-viewable-items-changed    on-viewable-items-changed
-       :on-end-reached               list-on-end-reached
-       :on-scroll-to-index-failed    identity ; don't remove this
-       :content-container-style      {:padding-top    (+ bottom-space 32)
-                                      :padding-bottom 16}
-       :scroll-indicator-insets      {:top bottom-space} ; iOS only
-       :keyboard-dismiss-mode        :interactive
-       :keyboard-should-persist-taps :handled
-       :onMomentumScrollBegin        state/start-scrolling
-       :onMomentumScrollEnd          state/stop-scrolling
-       :scrollEventThrottle          16
-       :on-scroll                    on-scroll
-       ;; TODO https://github.com/facebook/react-native/issues/30034
-       :inverted                     (when platform/ios? true)
-       :style                        (when platform/android? {:scaleY -1})
-       :on-layout                    on-messages-view-layout}]
-     [quo/floating-shell-button
-      (merge {:jump-to
-              {:on-press #(do
-                            (rf/dispatch [:chat/close true])
-                            (rf/dispatch [:shell/navigate-to-jump-to]))
-               :label    (i18n/label :t/jump-to)}}
-             (when @show-floating-scroll-down-button
-               {:scroll-to-bottom {:on-press scroll-to-bottom}}))
-      {:position :absolute
-       :bottom   6}]]))
+  [chat insets]
+  [:f>
+   (fn []
+     (let [keyboard-show-listener (atom nil)
+           keyboard-hide-listener (atom nil)
+           keyboard-shown         (atom false)]
+       (rn/use-effect
```

Comment:
I think we can do better with these effects mixed with the rest of the component code, particularly when the effect is self-contained like this one. The code below should work, but I haven't tested it.

My goal here was to help decouple the effectful code from the normal hiccup code. It's interesting to see the listener states don't leak to the rest of the app and are contained inside the hook. I've also renamed `:shown?` to use a question mark, since the value is a boolean.

```clojure
(defn use-keyboard-visibility
  []
  (let [show-listener (atom nil)
        hide-listener (atom nil)
        shown?        (atom nil)]
    (rn/use-effect
     (fn []
       (reset! show-listener
         (.addListener rn/keyboard "keyboardWillShow" #(reset! shown? true)))
       (reset! hide-listener
         (.addListener rn/keyboard "keyboardWillHide" #(reset! shown? false)))
       (fn []
         (.remove ^js @show-listener)
         (.remove ^js @hide-listener))))
    {:shown? shown?}))
```

And then in the component:

```clojure
(let [{keyboard-shown? :shown?} (use-keyboard-visibility)]
  [messages-list-content chat insets keyboard-shown?])
```

Reply:
Works well 🚀